product_packaging_level/models/product_packaging.py

```diff
@@ -28,15 +28,46 @@ class ProductPackaging(models.Model):
     qty_per_level = fields.Char(
         compute="_compute_qty_per_level", string="Qty per package level"
     )
+    name_type = fields.Selection(
+        string="Name Based On",
+        selection=[
+            ("by_package_level", "Package Level Name"),
+            ("by_package_type", "Package Type Name"),
+            ("user_defined", "User Defined"),
+        ],
+        default="by_package_level",
+        help=(
+            "config to set name of product packaging. Three options:"
+            "- The package level name (default)"
+            "- The package type name (if groups='stock.group_tracking_lot')"
+            "- user defined: free text value defined"
+        ),
+    )
```

Comment:
This config should be defined on the stock_package_level and not on the product_packaging
(so that we don't have to configure on each packaging and it will depend on which packaging level is chosen).

Reply:
updated